pkg/controller/infrastructure/infraflow/reconciler.go

```diff
@@ -235,3 +273,60 @@ func (fctx *FlowContext) loadWhiteBoard() {
 
 	fctx.whiteboard.Set(CreatedResourcesExistKey, "true")
 }
+
+func PatchProviderStatusAndState(
+	ctx context.Context,
+	runtimeClient client.Client,
+	infra *extensionsv1alpha1.Infrastructure,
+	networking *v1beta1.Networking,
+	status *v1alpha1.InfrastructureStatus,
+	state *runtime.RawExtension,
+	nodesSubnetIPv6CIDR *string,
+	servicesSubnetIPv6CIDR *string,
+) error {
+	patch := client.MergeFrom(infra.DeepCopy())
+
+	if status != nil {
+		infra.Status.ProviderStatus = &runtime.RawExtension{Object: status}
+		infra.Status.EgressCIDRs = make([]string, len(status.Networks.NatIPs))
+		for i, natIP := range status.Networks.NatIPs {
+			infra.Status.EgressCIDRs[i] = fmt.Sprintf("%s/32", natIP.IP)
+		}
```

Comment:
Probably this needs to be changed too. The ipv6 ranges should be added to this EgressCIDRs too I believe.